enderio-machines/src/main/java/com/enderio/machines/client/widget/EnergyWidget.java

```diff
@@ -0,0 +1,73 @@
+package com.enderio.machines.client.widget;
+
+import com.enderio.base.common.lang.EIOLang;
+import com.enderio.base.common.util.TooltipUtil;
+import com.enderio.base.common.util.Vector2i;
+import com.enderio.machines.EIOMachines;
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.PoseStack;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.components.AbstractWidget;
+import net.minecraft.client.gui.narration.NarrationElementOutput;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.client.renderer.GameRenderer;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.resources.ResourceLocation;
+
+import java.util.function.Supplier;
+
+public class EnergyWidget extends AbstractWidget {
+
+    private static final ResourceLocation WIDGETS = EIOMachines.loc("textures/gui/widgets.png");
+
+    private final Screen displayOn;
+    private final Supplier<Vector2i> getEnergy;
+
+    public EnergyWidget(Screen displayOn, Supplier<Vector2i> getEnergy, int pX, int pY, int pWidth, int pHeight) {
+        super(pX, pY, pWidth, pHeight, TextComponent.EMPTY);
+        this.displayOn = displayOn;
+        this.getEnergy = getEnergy;
+    }
+
+    @Override
+    public void updateNarration(NarrationElementOutput pNarrationElementOutput) {}
+
+    // Stop the click sound
+    @Override
+    public boolean mouseClicked(double pMouseX, double pMouseY, int pButton) {
+        return false;
+    }
+
+    @Override
+    public void renderButton(PoseStack pPoseStack, int pMouseX, int pMouseY, float pPartialTicks) {
+        Minecraft minecraft = Minecraft.getInstance();
+        RenderSystem.setShader(GameRenderer::getPositionTexShader);
+        RenderSystem.defaultBlendFunc();
+        RenderSystem.enableDepthTest();
+        RenderSystem.setShaderTexture(0, WIDGETS);
```

Comment:
maybe move rendersystem setup to some kind of GuiHelper in base, as we use that there for the EIOScreen as well

Reply:
This can be done in a cleanup PR sometime